third_party/move/move-compiler-v2/src/pipeline/unreachable_code_analysis.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Implements a data-flow analysis to determine whether an instruction is reachable or not.
+//! This analysis does not have any prerequisites.
+//! This analysis sets an annotation of type `UnreachableCodeAnnotation` on each function target.
+//! This annotation is a prerequisite for the unreachable code checker and unreachable code remover.
+//!
+//! This analysis a forward "may" analysis, it tracks whether an instruction is:
+//! - maybe reachable (there may be an execution path from the function entry to the instruction)
+//! - definitely not reachable (there is no execution path from the function entry to the instruction)
+
+use move_binary_format::file_format::CodeOffset;
+use move_model::model::FunctionEnv;
+use move_stackless_bytecode::{
+    dataflow_analysis::{DataflowAnalysis, TransferFunctions},
+    dataflow_domains::{AbstractDomain, JoinResult},
+    function_target::{FunctionData, FunctionTarget},
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    stackless_bytecode::Bytecode,
+    stackless_control_flow_graph::StacklessControlFlowGraph,
+};
+use std::collections::BTreeMap;
+
+/// Reachability state of an instruction.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum ReachableState {
+    Maybe, // Maybe reachable from function entry
+    No,    // Definitely not reachable from function entry
+}
+
+impl ReachableState {
+    /// Mark this state as maybe reachable from the function entry.
+    fn mark_as_maybe_reachable(&mut self) {
```

Comment:
Frankly, I think those setters are a bit of an overkill, specifically as they are module private. Its pretty idiomatic in Rust actually to work with open structs (enum and matching is based on this).

Reply:
Alright, I will get rid of them.

I wrote this thinking it would help with readability, but given their minimal use, I can go either way.